sympy/printing/tests/test_str.py

```diff
@@ -224,6 +227,14 @@ def test_Mul():
     assert str(-2*x/3) == '-2*x/3'
     assert str(-1.0*x) == '-1.0*x'
     assert str(1.0*x) == '1.0*x'
+    assert str(Mul(0, 1, evaluate=False)) == '1*0'
```

Comment:
Why is the order being reversed? This is different from the Add case. Does `Mul(1, 0, evaluate=0)` print the same?

Reply:
I don't normally mess around with the printing code so I'm not entirely sure of the reasons for this but Mul does not print its args in order in general:
https://github.com/sympy/sympy/blob/aa9c901819d9390d5ed13a269bfc32569ea749b6/sympy/printing/str.py#L290-L294
This means that (on master):
```julia
In [1]: m = Mul(y, x, 2, 1, evaluate=False)                                                                                       

In [2]: m.args                                                                                                                    
Out[2]: (y, x, 2)

In [3]: str(m)                                                                                                                    
Out[3]: '2*x*y'

In [4]: m                                                                                                                         
Out[4]: 2⋅x⋅y
```
This PR gives
```julia
In [2]: m.args                                                                                                                    
Out[2]: (y, x, 2, 1)

In [3]: str(m)                                                                                                                    
Out[3]: '1*2*x*y'

In [4]: m                                                                                                                         
Out[4]: 1⋅2⋅x⋅y
```
Various manipulations are done that combine rationals e.g. (on master and here):
```julia
In [7]: str(Mul(S(2)/3, S(3)/4, evaluate=False))                                                                                  
Out[7]: '2*3/(3*4)'

In [8]: str(Mul(2, S(1)/3, S(3)/4, evaluate=False))                                                                               
Out[8]: '3*2/(3*4)'
```
The changes to the printing code here are just to stop identities being absorbed into other rationals so that you can at least see that you have a funny kind of object. This previously printed as 1 which was confusing while debugging:
```julia
In [9]: Mul(1, 1, evaluate=False)                                                                                                 
Out[9]: 1⋅1

In [10]: Mul(1, 1, evaluate=False).args                                                                                           
Out[10]: (1, 1)
```